core/common/src/serialization_t.cpp

```diff
@@ -137,6 +139,147 @@ TEST_CASE("Serialization",
     const auto &s{m2.getSimulationSettings()};
     REQUIRE(s.options.pixel.maxErr.rel == dbl_approx(0.005));
   }
+  SECTION("Valid current v4 sme file. The number of colors equals the number "
+          "of sampledVolumes") {
+    // sme versions > 1.3.1
+    QFile f(":/models/very-simple-model.xml");
+    f.open(QIODevice::ReadOnly);
+    model::Model m;
+    m.importSBMLString(f.readAll().toStdString());
+    m.exportSMEFile("test2.sme");
+    model::Model m2;
+    m2.importFile("test2.sme");
+    const auto &s{m2.getSampledFieldColours()};
+    REQUIRE(s[0] == 4278190592);
+    REQUIRE(s[1] == 4287652289);
+    REQUIRE(s[2] == 4291134816);
+    const auto &s2{m2.getCompartments()};
+    REQUIRE(s2.getColours().size() == 3);
+    REQUIRE(s2.getColours()[0] == 4278190592);
+    REQUIRE(s2.getColours()[1] == 4287652289);
+    REQUIRE(s2.getColours()[2] == 4291134816);
+  }
+
+  SECTION("Valid current v4 sme file. The number of colors is greater than the "
+          "number of sampledVolumes") {
+    // sme versions > 1.3.1
+    QFile f(":/models/liver-simplified.xml");
+    f.open(QIODevice::ReadOnly);
+    model::Model m;
+    m.importSBMLString(f.readAll().toStdString());
+    m.exportSMEFile("test2.sme");
+    model::Model m2;
+    m2.importFile("test2.sme");
+    const auto &s{m2.getSampledFieldColours()};
+    REQUIRE(s[0] == 4278719745);
+    REQUIRE(s[1] == 4278237440);
+    REQUIRE(s[2] == 4279380443);
+    const auto &s2{m2.getCompartments()};
+    REQUIRE(s2.getColours().size() == 2);
+    REQUIRE(s2.getColours()[0] == 4278237440);
+    REQUIRE(s2.getColours()[1] == 4279380443);
+  }
+
+  SECTION("Import a model saved using the old (1.3.1) format.") {
+    // sme versions > 1.3.1
+    QFile f(":test/models/liver-simplified_v1.3.1.xml");
+    f.open(QIODevice::ReadOnly);
+    model::Model m;
+    m.importSBMLString(f.readAll().toStdString());
+    const auto &s{m.getCompartments()};
+    REQUIRE(s.getColours().size() == 2);
+    REQUIRE(s.getColours()[0] == 4278237440);
+    REQUIRE(s.getColours()[1] == 4279380443);
+    m.exportSMEFile("test3.sme");
+
+    model::Model m2;
+    m2.importFile("test3.sme");
```

Comment:
Here it would be nice to check that the compartment geometry images are the same for `m` and `m2`, e.g.
```
    REQUIRE(m.getGeometry().getImage() == m2.getGeometry().getImage());
```
And also check that the compartment pixels are the same, e.g.
```
    REQUIRE(s.getCompartments()[0]->nPixels() == s3.getCompartments()[0]->nPixels());
    REQUIRE(s.getCompartments()[1]->nPixels() == s3.getCompartments()[1]->nPixels());
```